File: api/src/modules/auth/dto/register-request.dto.ts
import { Transform } from 'class-transformer';
import { IsEmail, IsNotEmpty, MinLength } from 'class-validator';
import { lowerCaseTransformer } from 'src/common/transformers/lower-case.transformer';

export class RegisterRequestDto {
  @Transform(lowerCaseTransformer)
  @IsEmail({}, { message: 'Provided value is invalid email' })
  email: string;

  @MinLength(6, { message: 'Password should have at least 6 characters' })
  password: string;

  @IsNotEmpty({ message: 'Firstname is missing' })
  firstName: string;

  @IsNotEmpty({ message: 'Lastname is missing' })
  lastName: string;
}

File: app/lib/models/request/register_request_dto.dart
class RegisterRequestDto {
  String firstName;
  String lastName;
  String email;
  String password;

  RegisterRequestDto({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.password,
  });

  Map<String, dynamic> toJson() => {
        "firstName": firstName,
        "lastName": lastName,
        "email": email,
        "password": password,
      };
}
